core/internal/cltest/cltest.go

```diff
@@ -161,12 +162,33 @@ func NewApplication() (*TestApplication, func()) {
 	}
 }
 
+// NewApplicationWithKey creates a new TestApplication along with a new config
+func NewApplicationWithKey() (*TestApplication, func()) {
+	config, cfgCleanup := NewConfig()
+	app, cleanup := NewApplicationWithConfigAndKey(config)
+	return app, func() {
+		cleanup()
+		cfgCleanup()
+	}
+}
+
+// NewApplicationWithConfigAndKey creates a new TestApplication with the given testconfig
+// it will also provide an unlocked account on the keystore
+func NewApplicationWithConfigAndKey(tc *TestConfig) (*TestApplication, func()) {
+	app, cleanup := NewApplicationWithConfig(tc)
+	app.ImportKey(key3cb8e3fd9d27e39a5e9e6852b0e96160061fd4ea)
+	return app, cleanup
+}
+
 // NewApplicationWithConfig creates a New TestApplication with specified test config
 func NewApplicationWithConfig(tc *TestConfig) (*TestApplication, func()) {
 	WipePostgresDatabase(tc.Config)
-	app := services.NewApplication(tc.Config).(*services.ChainlinkApplication)
+	ta := &TestApplication{connectedChannel: make(chan struct{}, 1)}
+	app := services.NewApplication(tc.Config, func(app services.Application) {
+		ta.connectedChannel <- struct{}{}
```

Comment:
These few lines seem to be the prize.

1. Create a `connectChannel` on the instance.
2. Create a lambda function to signal when connected.
3. Feed it into the `NewApplication` constructor so that it has the callback from instantiation rather than an attachment happening at runtime.
4. Application calls all `HeadTrackable` callbacks as always.

👍